src/js_sync.hpp

```diff
@@ -977,8 +976,8 @@ void SyncClass<T>::populate_sync_config(ContextType ctx, ObjectType realm_constr
         if (!(Object::template is_instance<UserClass<T>>(ctx, user_object))) {
             throw std::invalid_argument("Option 'user' is not a Realm.User object.");
         }
-        SharedUser user = *get_internal<T, UserClass<T>>(ctx, user_object);
-        if (user->state() != SyncUser::State::LoggedIn) {
+        auto user = get_internal<T, UserClass<T>>(ctx, user_object);
```

Comment:
[non-blocking]  Is there a reason the type here has been changes from `SharedUser` to `auto`?  IMHO, `SharedUser` makes it easier to read/debug.

Reply:
It's not a `SharedUser` anymore.  `user->m_user` is.